### DISCUSSION TITLE
infracost to support configuring estimates for unsupported resources

### DISCUSSION THREAD

**Discussion by moreandres** [infracost/infracost#1271] [Q&A]
Hi! Is there an infracost feature already to add to the YAML configuration file manually-provided monthly estimates for unsupported resources? For instance  if `show-skipped` shows `14 x aws_glue_crawler`, I would like to configure `aws_glue_crawler` to some monthly estimated figure myself until proper support is in there. Thanks!

**Comment by alikhajeh1** [comment]
@moreandres that's a cool idea! We don't have that feature. Would you get that figure from AWS Cost Explorer? Are you trying to make the monthly cost estimate shown in Infracost becomes closer to the actual costs for that project or account as shown in AWS Cost Explorer?

Which option would you find more useful?

### Option 1: total cost of the AWS Glue service

Usage file:
```
aws_glue:
  monthly_cost_dollars: 6000
```

`infracost breakdown` output:
```
 Name                        Price    Monthly Qty   Unit    Hourly Cost  Monthly Cost

 AWS Glue
 └─ Monthly cost             $0.77    1             USD     $8.22        $6000
```

### Option 2: cost of individual AWS Glue crawler resources

Usage file:
```
  aws_glue_crawler.my_glue_1:
    monthly_cost_dollars: 6000
```

`infracost breakdown` output:
```
 Name                        Price    Monthly Qty   Unit    Hourly Cost  Monthly Cost

 aws_glue_crawler.my_glue_1
 └─ Monthly cost             $0.77    1             USD     $8.22        $6000
```

**Reply by moreandres** [reply]
Thanks for the prompt response! 

I would say the second option of being able to add a per resource monthly estimate fits better, as `show-skipped` will list the resource you need to add into the configuration.

**Reply by alikhajeh1** [reply]
Thanks @moreandres! If we do https://github.com/infracost/infracost/issues/1272, do you still need this feature?

**Reply by moreandres** [reply]
I would say yes, as a fallback option for resources not related to Glue.

**Reply by alikhajeh1** [reply]
Makes sense! So to be clear, your ultimate aim is to make the monthly cost estimate shown in Infracost get closer to the actual costs shown in AWS Cost Explorer? Are you using AWS tags to get the cost for the Terraform project?

Also, I'm also curious if you want this because the people seeing the cost estimate comment don't have access to the AWS Cost Explorer?

**Comment by gysiedebruyn** [comment]
### Cost diff discrepancy on Azure Event Hub service
Azure Event Hub change Standard to Premium SKU and Infracost shows decrease in price, pretty sure the increase is actually quite big. Result of diff:
```
Detected Terragrunt directory at workloads/development/australiaeast/eventhub
  ✔ Running terragrunt run-all terragrunt-info 
  ✔ Running terragrunt run-all plan 
  ✔ Running terragrunt show 
  ✔ Extracting only cost-related params from terragrunt plan 
  ✔ Retrieving cloud prices to calculate costs 

Project: gysiedebruyn/terraform-azure/workloads/development/australiaeast/eventhub

~ azurerm_eventhub_namespace.ns
  -$21.90 ($21.90 → $0.00)

    - Ingress event (Standard)
      Monthly cost depends on usage
        -$0.028 per 1M events

    ~ Throughput
      -$21.90 ($21.90 → $0.00)

Monthly cost change for gysiedebruyn/terraform-azure/workloads/development/australiaeast/eventhub
Amount:  -$21.90 ($29.70 → $7.80)
Percent: -74%

──────────────────────────────────
Key: ~ changed, + added, - removed

4 cloud resources were detected, rerun with --show-skipped to see details:
∙ 3 were estimated, 2 include usage-based costs, see https://infracost.io/usage-file
∙ 1 was free
```

This is the actual Terraform plan output:
```
  # azurerm_eventhub_namespace.ns must be replaced
-/+ resource "azurerm_eventhub_namespace" "ns" {
      ~ default_primary_connection_string         = (sensitive value)
      + default_primary_connection_string_alias   = (sensitive value)
      ~ default_primary_key                       = (sensitive value)
      ~ default_secondary_connection_string       = (sensitive value)
      + default_secondary_connection_string_alias = (sensitive value)
      ~ default_secondary_key                     = (sensitive value)
      ~ id                                        = "/.../jdb-tf-standard-hub" -> (known after apply)
      ~ maximum_throughput_units                  = 0 -> (known after apply)
        name                                      = "jdb-tf-standard-hub"
      ~ network_rulesets                          = [
          ~ {
              ~ virtual_network_rule           = [] -> null
                # (3 unchanged elements hidden)
            },
        ]
      ~ sku                                       = "Standard" -> "Premium" # forces replacement
        tags                                      = {
            "CostCentre" = "PoC"
            "Module"     = "eventhub"
            "Owner"      = "Jan de Bruyn"
        }
      ~ zone_redundant                            = false -> true # forces replacement
        # (4 unchanged attributes hidden)
    }
```

**Reply by tim775** [reply]
Thanks for the issue!  I created #1354 to track this.

**Reply by tim775** [reply]
The fix for this will go out in the v0.9.18 release.  If you get a chance, take a look at [the test output](https://github.com/infracost/infracost/blob/ab26b2332e67fbe8b5ec6e446c56f2b5f82df7a4/internal/providers/terraform/azure/testdata/event_hubs_namespace_test/event_hubs_namespace_test.golden#L20-L26) to make sure it's inline with what you expect. Thanks again!
